tests/unit/test_flight.py

```diff
@@ -816,6 +817,113 @@ def test_antimeridian_long_cross(direction: str, cross_anti: bool) -> None:
+        latitude=np.zeros(n),
+        altitude=np.full(n, 10000),
+        time=pd.date_range("2022-01-01", "2022-01-01T06", n),
+    )
+    fl2 = fl.resample_and_fill("5s")
+    assert np.all(fl2["longitude"] < 180)
+    assert np.all(fl2["longitude"] >= -180)
+    assert np.all(np.isfinite(fl2["longitude"]))
+    assert np.all(fl2.segment_length()[:-1] < 10000)
+    assert np.all(fl2.segment_length()[:-1] > 8000)
+
+
+@pytest.mark.parametrize("direction", ["east", "west"])
+def test_meridian_antimeridian_cross(direction: str) -> None:
+    """Test interpolation adjustments for antimeridian crossing.
+
+    This test uses longitude values that span both the meridian and antimeridian.
+    """
+    n = 200
+    if direction == "east":
+        longitude = np.linspace(-10, -10 + 340, n)
+    elif direction == "west":
+        longitude = np.linspace(10, 10 - 340, n)
+    else:
+        raise ValueError("Unknown param")
+    longitude = (longitude + 180) % 360 - 180
+    fl = Flight(
+        longitude=longitude,
+        latitude=np.zeros(n),
+        altitude=np.full(n, 10000),
+        time=pd.date_range("2022-01-01", "2022-01-01T06", n),
+    )
+    fl2 = fl.resample_and_fill("5s")
+    assert np.all(fl2["longitude"] < 180)
+    assert np.all(fl2["longitude"] >= -180)
+    assert np.all(np.isfinite(fl2["longitude"]))
+    assert np.all(fl2.segment_length()[:-1] < 10000)
+    assert np.all(fl2.segment_length()[:-1] > 8000)
+
+
+@pytest.mark.parametrize("direction", ["east", "west"])
+def test_consecutive_antimeridian_cross(direction: str) -> None:
+    """Test error on consecutive antimeridian crossings in the same direction."""
+    n = 100
+    if direction == "east":
+        longitude = np.linspace(90, 90 + 540, n)
+    elif direction == "west":
+        longitude = np.linspace(-90, -90 - 540, n)
+    else:
+        raise ValueError("Unknown param")
+    longitude = (longitude + 180) % 360 - 180
+    fl = Flight(
+        longitude=longitude,
+        latitude=np.zeros(n),
+        altitude=np.full(n, 10000),
+        time=pd.date_range("2022-01-01", "2022-01-01T10", n),
+    )
+    with pytest.raises(ValueError, match="Cannot handle consecutive"):
+        fl.resample_and_fill("5s")
+
+
+@pytest.mark.parametrize("direction", ["east", "west"])
+def test_no_viable_antimeridian_shift(direction: str) -> None:
+    """Test error for resampling flights that span 360 degrees longitude."""
+    n = 100
+    if direction == "east":
+        longitude = np.linspace(0, 360, n)
+    elif direction == "west":
+        longitude = np.linspace(0, -360, n)
+    else:
+        raise ValueError("Unknown param")
+    longitude = (longitude + 180) % 360 - 180
+    fl = Flight(
+        longitude=longitude,
+        latitude=np.zeros(n),
+        altitude=np.full(n, 10000),
+        time=pd.date_range("2022-01-01", "2022-01-01T06", n),
+    )
+    with pytest.raises(ValueError, match="Cannot handle flight that spans"):
+        fl.resample_and_fill("5s")
+
+
```

Comment:
Is it worth having a test for a flight that zippers back and forth across the anti-meridian? Something like

```
longitude = (np.sin(np.linspace(0, 20, 1000))) % 360.0 - 180.0
```

I would expect a flight with this longitude field to make it through `Flight.resample_and_fill` without any errors.

Reply:
https://github.com/contrailcirrus/pycontrails/pull/222/commits/2b783e24d8378aeab1e4f117b6d15b493e9bc274